6657

OLGKAD opened the pull request:
Fixes #

**PR Checklist**

<!-- Remove this portion after you have made the checks. -->

Ensure that you have:
- [ ] Read and understood our PR guideline: https://github.com/TEAMMATES/teammates/blob/master/docs/process.md#step-4-submit-a-pr
  - [ ] Added the issue number to the "Fixes" keyword above
  - [ ] Titled the PR as specified in the abovementioned document
- [ ] Made your changes on a branch other than `master` and `release`
- [ ] Gone through all the changes in this PR and ensured that:
  - [ ] They addressed one (and only one) issue
  - [ ] No unintended changes were made
- [ ] Run and passed static analysis: `./gradlew lint` and `npm run lint`
- [ ] Added/updated tests, if changes in functionality were involved
- [ ] Added/updated documentation to public APIs (classes, methods, variables), if applicable

**Outline of Solution**

<!-- Tell us how you solved the issue. -->
<!-- If there are things you want the reviewers to focus on, include them here as well. -->
<!-- This portion can be skipped if the fix is trivial. -->

OLGKAD committed:
refactor: #6657: getQuestionResultStatisticsCsv

A chunk of code appending a string to the csvBody in the for loop is
now a separate function called updateQuestionResultStatisticsCsv().

OLGKAD committed:
refactor: #3 getQuestionResultStatisticsCsv

Added a function to create a custom decimal format.

OLGKAD committed:
refactor: #3 getQuestionResultStatisticsCsv

Modified updateQuestionResultStatisticsCsv to accept a slightly
different set of inputs.

Lucasljungberg commented on src/main/java/teammates/common/datatransfer/questions/FeedbackNumericalScaleQuestionDetails.java:
```diff
@@ -248,7 +248,17 @@ private String getInstructorQuestionResultsStatisticsHtml(
                         Slots.STATS_FRAGMENTS, fragmentHtml.toString());
     }
-
+    ////////////////////////////////////////////////////////////////////
```
Are these lines of comments an accident?

Lucasljungberg commented on src/main/java/teammates/common/datatransfer/questions/FeedbackNumericalScaleQuestionDetails.java:
```diff
@@ -265,8 +275,11 @@ private String getStudentQuestionResultsStatisticsHtml(
         // need to know which recipients are hidden since anonymised recipients will not appear in the summary table
         List<String> hiddenRecipients = getHiddenRecipients(responses, question, bundle);
 
+        System.out.println("Called function");
```
Try to remove some outputs

Lucasljungberg commented on src/main/java/teammates/common/datatransfer/questions/FeedbackNumericalScaleQuestionDetails.java:
```diff
@@ -447,6 +460,17 @@ private String getAverageExcludingSelfText(boolean showAvgExcludingSelf, Decimal
         return "";
     }
 
+////////////////////////////////////////////////////////////////////
```
If these aren't mean to be here, you should remove them too.

Lucasljungberg commented on src/main/java/teammates/common/datatransfer/questions/FeedbackNumericalScaleQuestionDetails.java:
```diff
@@ -528,7 +573,17 @@ private boolean showAverageExcludingSelf(
         return false;
     }
-
+    ////////////////////////////////////////////////////////////////////
```
If these aren't mean to be here, you should remove them too.

Lucasljungberg commented on src/main/java/teammates/common/datatransfer/questions/FeedbackNumericalScaleQuestionDetails.java:
```diff
@@ -537,7 +592,9 @@ private void populateSummaryStatisticsFromResponses(
             Map<String, Integer> numResponses,
             Map<String, Integer> numResponsesExcludingSelf) {
 
+        System.out.println("ALSKDJ");
```
You should remove the outputs (it wasn't you who put them there though, but it's still good to do it ^^ )

Lucasljungberg commented on src/main/java/teammates/common/datatransfer/questions/FeedbackNumericalScaleQuestionDetails.java:
```diff
@@ -599,7 +658,9 @@ private void populateSummaryStatisticsFromResponses(
         FeedbackParticipantType type = question.recipientType;
+
         for (FeedbackResponseAttributes response : responses) {
+            if (!bundle.visibilityTable.containsKey(response.getId())) continue;
```
I think this part is handled by another branch.

OLGKAD replied:
Thanks for the comments! Should be fixed now.

Lucasljungberg reviewed (changes requested):
Review the comments.

OLGKAD committed:
refactor: #4 getStudentQuestionResultsStatisticsHtml

Moved a chuck of code that produced recipientFragmentHtml by updating
fragmentTemplateToUse with a single record into a separate function.
